prov/util/src/util_attr.c

```diff
@@ -1002,6 +1002,17 @@ int ofi_check_tx_attr(const struct fi_provider *prov,
 	return 0;
 }
 
+int ofi_check_hmem_attr(const struct fi_provider *prov,
+			const struct fi_info *user_info)
+{
+	if (!(user_info->caps & FI_HMEM)) {
+		FI_INFO(prov, FI_LOG_CORE, "FI_HMEM not set\n");
+		return -FI_ENODATA;
+	}
```

Comment:
Do you need to check whether the opt in the hints can be satisfied by the provider? Like `ofi_check_tx_attr`